frontend/apps/filemanager/filemanagerbookinfo.lua

```diff
@@ -163,24 +164,11 @@ function BookInfo:show(file, book_props, metadata_updated_caller_callback)
             self.custom_doc_settings = nil
             self.custom_book_cover = nil
             if self.prop_updated then
-                local ui, fm_ui
-                if self.ui then
-                    if self.prop_updated == "title" then
-                        self.ui.view.footer:updateFooterText() -- in case the title changed
-                    end
-                else
-                    fm_ui = require("apps/filemanager/filemanager").instance
-                end
-                ui = self.ui or fm_ui
+                local ui = self.ui or require("apps/filemanager/filemanager").instance
                 if ui.coverbrowser then -- refresh cache db
                     ui.coverbrowser:deleteBookInfo(file)
                 end
-                if fm_ui then
-                    fm_ui:onRefresh()
-                end
-                if metadata_updated_caller_callback then
-                    metadata_updated_caller_callback()
-                end
+                UIManager:broadcastEvent(Event:new("BookMetadataChanged", self.prop_updated))
```

Comment:
Ok with the idea.
But `self.prop_updated` being just one single string ("title", "authors") might be enough for what you tweaked in this PR (refreshing FM & History, updating title in footer), but won't be for statistics (and for possibly replacing the `ui.coverbrowser:deleteBookInfo(file)` just above).

Reply:
deleteBookInfo must be done before other event handlers, so I left it here.

Reply:
This event is triggered on updating one property only.

Reply:
So ? Will there be another event introduced?
May be I don't get what are your plans with this PR :)

When updating a book multiple metadata, is it saved in one batch ? Or is each individual metatadata saved individually when it is edited?
If it is each individually, I guess your event is fine. It just needs to pass as much information about the book as possible, so handlers can use what they need to update stuff on their side.
I guess, other than the metadata name you already have, it should have:
- previous metadata value (for statistics to find what to update - it may need other unchanged metadata too :/)
- new metadata value
- filepath (ie. if CoverBrowser would deleteBookInfo - ok, I get it that it should happen before) as other modules/plugins may use it
- the md5sum from the DocSettings ? (for statistics :/)

Or, I dunno, may be the filepath is enough for Statistics to open the DocSettings and get md5 and custom metadata (but it would need to get the previous metadata value too).